src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```diff
@@ -72,67 +83,58 @@
-      account.authenticate();
-      httpCon = createConnection(account, url, contentType, metadata);
-      mOutputStream  = httpCon.getOutputStream();
-      LOG.debug("Second attempt PUT {} successfull. Got output stream", url.toString());
-      mHttpCon = httpCon;
-    } catch (IOException e) {
-      LOG.error(e.getMessage());
-      throw e;
-    }
-  }
-
-  /**
-   * Creates HTTP Connection
-   *
-   * @param account JOSS Account
-   * @param url URL to the object
-   * @param contentType object content type
-   * @param metadata user provided meta data
-   * @return HttpURLConnection if success
-   * @throws IOException if error
-   */
-  private HttpURLConnection createConnection(JossAccount account, URL url, String contentType,
-      Map<String, String> metadata) throws IOException {
-    LOG.debug("Create chunked HTTP connection for PUT {}. Read timeout {}. Streaming chunk {}",
-        url.toString(), READ_TIMEOUT, STREAMING_CHUNK);
-    HttpURLConnection newHttpCon = (HttpURLConnection) url.openConnection();
-    newHttpCon.setDoOutput(true);
-    newHttpCon.setRequestMethod("PUT");
-
-    newHttpCon.addRequestProperty("X-Auth-Token",account.getAuthToken());
-    newHttpCon.addRequestProperty("Content-Type", contentType);
+    mAccount = account;
+    client = connectionManager.createHttpConnection();
+    request = new HttpPut(mUrl.toString());
+    request.addHeader("X-Auth-Token", account.getAuthToken());
     if (metadata != null && !metadata.isEmpty()) {
       for (Map.Entry<String, String> entry : metadata.entrySet()) {
-        newHttpCon.addRequestProperty("X-Object-Meta-" + entry.getKey(), entry.getValue());
+        request.addHeader("X-Object-Meta-" + entry.getKey(), entry.getValue());
+      }
+    }
+
+    PipedOutputStream out = new PipedOutputStream();
+    final PipedInputStream in = new PipedInputStream();
+    out.connect(in);
+    mOutputStream = out;
+    writeThread = new Thread() {
+      public void run() {
+        InputStreamEntity entity = new InputStreamEntity(in, -1);
+        entity.setChunked(true);
+        entity.setContentType(contentType);
+        request.setEntity(entity);
+        try {
+          HttpResponse response = client.execute(request);
+          int responseCode = response.getStatusLine().getStatusCode();
+          if (responseCode >= 400) {
+            LOG.warn("Http Error Code: {} for {}, \nReason: {}", responseCode, mUrl.toString(),
+                    response.getStatusLine().getReasonPhrase());
+            mAccount.authenticate();
+            request.removeHeaders("X-Auth-Token");
+            request.addHeader("X-Auth-Token", mAccount.getAuthToken());
+            LOG.warn("Token recreated for {}.  Retry request", mUrl.toString());
+            response = client.execute(request);
+            responseCode = response.getStatusLine().getStatusCode();
+          }
+        } catch (IOException e) {
+          LOG.error(e.getMessage());
```

Comment:
@gilv This code executes the http request and starts the first time data is written to the stream.
I went with this route thinking it might be possible to use multiple threads for writing when we partition a large file. I haven't tested this yet but will work on it this week.
Also I'll add in exception handling by today or tomorrow.